### ISSUE TITLE
DX9 Eject & Reset crash

### ISSUE THREAD

**Issue by AyserJamshidi** [veeenu/hudhook#72]
When utilizing ImguiDx9Hooks, calling eject will cause a crash as the ImguiRenderer cleanup does nothing.  This can be slightly remedied by implementing it as:
```rust
unsafe fn cleanup(&mut self) {
    #[cfg(any(target_arch = "aarch64", target_arch = "x86_64"))]
    SetWindowLongPtrA(self.renderer.get_hwnd(), GWLP_WNDPROC, self.wnd_proc as usize as isize);

    #[cfg(target_arch = "x86")]
    SetWindowLongA(self.renderer.get_hwnd(), GWLP_WNDPROC, self.wnd_proc as usize as i32);
}
```

Note that even with this cleanup implementation **sometimes** ejecting (in a previously injected game process) will still cause a crash, but not every time.  This happens even when utilizing the hudhook! macro and using a modified version of the README ImGui window that contains only a button that calls eject.

This is not a whole solution as DX9 will still crash when/after Reset gets called.  I'm unsure if it's directly caused by Reset, but when resizing a game window (Aion MMO in this case), it will call Reset, which promptly crashes in the sequence of: 
* Cursor starts resizing Game window by dragging the window border
* Game turns to black due to resize (this is normal behavior for this game)
* Game renders again with the ImGui overlay for one frame
* Game freezes/crashes

I have not made a PR for this fix due to the Reset issue, but implementing cleanup and waiting a bit before calling eject does improve the DX9 experience.  

If any extra information is needed on these issues please let me know.  Thank you

**Comment by veeenu** [comment]
Thank you for the feedback!

I'm assuming adding some sort of [synchronization](https://github.com/veeenu/hudhook/blob/main/src/hooks/dx12.rs#L800-L828) may be in order, to make sure we don't hard kill stuff that needs proper cleanup. That could be the culprit for the remainder of crashes, but resetting the correct window proc is already a good step towards that.

Feel free to submit a PR -- it's ok if issues get fixed gradually!

**Comment by veeenu** [comment]
Closing this as most likely fixed by #74 -- we need more evidence to make sure, but we can reopen if the issue persists.
